src/ffi/mod.rs

```diff
@@ -610,12 +677,17 @@ pub extern "C" fn telio_enable_magic_dns(
     dev: &telio,
     forward_servers: *const c_char,
 ) -> telio_result {
+    let servers_str = ffi_try!(unsafe { CStr::from_ptr(forward_servers) }
```

Comment:
do we know that `forward_servers` is not null  here?

Reply:
We don't. I had just changed the order. Added a check.